### PR TITLE
Fixed broken build on 32-bit systems

### PR DESCRIPTION
**Pull request by chenguokai**
Fixes https://github.com/stlink-org/stlink/issues/919

### CODE REVIEW

File: src/common.c [hunk 1]
```diff
@@ -437,7 +437,7 @@ static void set_flash_cr_pg(stlink_t *sl) {
         x |= 1 << FLASH_CR_PG;
     } else if (sl->flash_type == STLINK_FLASH_TYPE_L4) {
         cr_reg = STM32L4_FLASH_CR;
-        x &= ~STM32L4_FLASH_CR_OPBITS;
+        x &= (uint32_t)~STM32L4_FLASH_CR_OPBITS;
         x |= 1 << STM32L4_FLASH_CR_PG;
```
Comment (slyshykO) [inline src/common.c:441]: Maybe it is more useful to make definition of STM32L4_FLASH_CR_OPBITS typed?
For example like this : 
```cpp
#define STM32L4_FLASH_CR_OPTLOCK  30u	/* Lock option bytes */
```

or

```cpp
#define STM32L4_FLASH_CR_OPTLOCK  (uint32_t)30	/* Lock option bytes */
```

File: src/common.c [hunk 2]
```diff
@@ -2000,7 +2000,7 @@ int stlink_erase_flash_mass(stlink_t *sl) {
     if (sl->flash_type == STLINK_FLASH_TYPE_L0 ||
         sl->flash_type == STLINK_FLASH_TYPE_WB) {
         /* erase each page */
-        int i = 0, num_pages = (int) sl->flash_size/sl->flash_pgsz;
+        int i = 0, num_pages = (int) sl->flash_size/(int)(sl->flash_pgsz);
```
Comment (slyshykO) [inline src/common.c:2003]: Think brackets should help there.
```cpp
int i = 0, num_pages = (int) (sl->flash_size / sl->flash_pgsz);
```

File: src/common.c (hunk 1 above)
Comment (chenguokai) [reply src/common.c]: Indeed better, changed

**Comment by slyshykO** [comment]
I think we should add such configuration to Travis build.
I mean configuration that causes issues, that this PR try to fix.

**Comment by slyshykO** [comment]
I think this config should be 64bit Linux with 32bit clang.

**Comment by Nightwalker-87** [comment]
Yes @slyshykO, it should be regularly tested with Travis. I agree on that.

**Comment by Nightwalker-87** [comment]
Thanks for fixing this! :+1: 

To be honest, we should change some variables to the appropriate sizes / datatypes. The requirement for manual casts not only here, but at several places throughout the codebase, indicates that something is not as it is supposed to be per design. If implemented nicely, there hardly is any need for explicit casting. I'd like to see that addressed in v1.6.2 at a later point in time, maybe within #909.

**Pull request merged**